File: archive/destination-city.js
const destCity = function(paths) {
  let mem = {}
  let s = new Set()
  paths.forEach(arr => {
    s.add(arr[0])
    s.add(arr[1])
    mem[arr[0]] = arr[1]
  })
  let ans = ''
  s.forEach(city => {
    if (!(city in mem)) {
      ans = city
    }
  })
  return ans
};

console.log(destCity([["London","New York"],["New York","Lima"],["Lima","Sao Paulo"]]))

File: go-archive/destination-city.go
func destCity(paths [][]string) string {
    mem:=map[string]string{}

    for _, k := range paths {
        mem[k[0]] = k[1]
    }

    return rec(paths[0][0], mem)
}

func rec(from string, mem map[string]string) string {
    if _, isHas := mem[from]; !isHas {return from}

    return rec(mem[from], mem)
}
